inspirehep/modules/workflows/tasks/matching.py

```diff
@@ -212,6 +212,28 @@ def _is_core(record):
         obj.extra_data['core'] = True
 
 
+def get_wfs_not_completed(obj, skip_blocked=True):
+    """Return list of all matched workflows which are not completed
+    By default it do not show workflows which are already blocked by this workflow
+
+    Arguments:
+        obj: a workflow object.
+        skip_blocked: boolean, if True do not returnd workflows blocked by this one,
+            if False then return all matched workflows.
+    """
+
+    def _non_completed(base_record, match_result):
+        return not get_value(match_result,
+                             '_source._workflow.status') == 'COMPLETED'
+
+    matched_ids = pending_in_holding_pen(obj, _non_completed)
+    if skip_blocked:
+        matched_ids = remove_blocked_by_me(obj, matched_ids)
```

Comment:
`matched_ids = [id_ for id_ in matched_ids if not is_workflow_id_blocked_by_workflow(obj, id_)]` (see further comment).